soc/xtensa/intel_adsp/ace_v1x/power.c

```diff
@@ -26,17 +40,97 @@
  * (each bit corresponds to one ebb)
  * @param response_to_ipc       flag if ipc response should be send during power down
  */
-extern void ace_power_down(bool disable_lpsram, uint32_t *hpsram_pg_mask,
+extern void power_down(bool disable_lpsram, uint32_t *hpsram_pg_mask,
 			   bool response_to_ipc);
 
+/* NOTE: This struct will grow with all values that have to be stored for
+ * proper cpu restore after PG.
+ */
+struct core_state {
+	uint32_t a0;
+	uint32_t a1;
+	uint32_t vecbase;
+	uint32_t excsave2;
+	uint32_t excsave3;
+	uint32_t thread_ptr;
+	uint32_t intenable;
+};
+
+static struct core_state core_desc[CONFIG_MP_NUM_CPUS] = { 0 };
+static struct core_state *desc_alias = &core_desc;
```

Comment:
What is the purpose of this alias ?

Reply:
I have not found a better solution. I'm using it to get struct address after core reset.